crudd/src/main.rs

```diff
@@ -412,28 +455,35 @@ async fn cmd_write<T: BlockIO>(opt: &Opt, crucible: Arc<T>) -> Result<()> {
                 }
             }
         }
+        total_bytes_written += n_read;
+
         if n_read < w_buf.capacity() {
             eprintln!("n_read was {}, returning early", n_read);
-            return write_remainder_and_finalize(
+            write_remainder_and_finalize(
                 &crucible,
                 w_buf,
                 offset,
                 n_read,
                 native_block_size,
                 futures,
             )
-            .await;
+            .await?;
+            return Ok(total_bytes_written);
         } else {
-            eprintln!("writing full iocmd");
             // good to go for a write
             let w_future = crucible.write(offset, w_buf.freeze());
             futures.push_back(w_future);
         }
 
-        // Block on futures so we dont get a backlog
+        // Block on oldest future so we dont get a backlog
```

Comment:
I'm not sure it matters here, but the oldest future might not be the first one to finish.
We have recently relaxed the forced job ordering, and it's possible that jobs submitted
will not finish in the order of submission.

Reply:
Oooh interesting. Yeah maybe we can switch this up to wait on first-available instead.